interfaces.go

```diff
@@ -5,14 +5,31 @@
 package optimize
 
 // A Method can optimize an objective function.
+//
+// It uses a reverse-communication interface design represented by the Init()
+// and Iterate() methods to pass and retrieve information from the client. Its
+// aim is to isolate the method from user-supplied information and data and
```

Comment:
I changed it to:

```
// This provides independence of the optimization algorithm on user-supplied
// data and their representation, and enables automation of common operations
// like checking for (various types of) convergence and maintaining statistics.
```

I wanted to keep the mention of user data and their representation. With Local() we make and fix all the choices for the user, but there is still the freedom to take a Method and use it in a custom driver routine with different representation of the problem, settings, convergence check, anything.